### PR TITLE
Show a success message when restoring a revision

### LINKED ISSUE #12618: Show a success message when restoring a revision
## Feature Description

Right now, restoring a revision doesn't display a success message.

The `story restored to revision from...` message isn't shown when restoring an older revision of a post. The message correctly appears when using the classic editor.

It's tricky to do this because the restoring actually happens on the server side, so the only indication we have is the existence of the `revision` GET parameter in the URL.

I think we could solve this by checking for that in `Editor` and passing a custom message string to the editor in `get_editor_settings()`.

Then, in the editor we have a simple component that checks for this message and displays it in a snackbar if it's there.

See Gutenberg PR below for inspiration.

## Additional Context

Gutenberg currently has the same issue, FWIW:

* https://github.com/WordPress/gutenberg/issues/19226
* https://github.com/WordPress/gutenberg/pull/20349

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Show a basic snackbar message on restoring a revision. 

## Testing Instructions

https://user-images.githubusercontent.com/237508/203111694-e4d251fd-3936-42b2-b3d2-a052fefc5a0d.mov

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Create a story. 
2. Publish a story. 
3. Make an edit. 
4. Update story. 
5. Go to document tab. 
6. Click revisions. 
7. Change story and restore story. 
8. See message. 

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12618

### CODE REVIEW

File: includes/Admin/Editor.php [hunk 1]
```diff
@@ -387,6 +387,20 @@ public function get_editor_settings(): array {
...
+		 *
+		 * @var int $revision
+		 */
+		$revision = isset( $_GET['revision'] ) ? absint( $_GET['revision'] ) : 0; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+		$revision_message = ! empty( $revision ) ?
+			sprintf(
+				/* translators: %s: Number of stories. */
+				__( 'Post restored to revision from %s.', 'web-stories' ),
+				wp_post_revision_title( $revision, false )
```
Comment (spacedmonkey) [inline includes/Admin/Editor.php:401]: It is a little hard to pass this in the rest api. So cleaner to just pass the whole message down.

File: includes/Admin/Editor.php [hunk 2]
```diff
@@ -387,6 +387,20 @@ public function get_editor_settings(): array {
...
+		 * Revision.
+		 *
+		 * @var int $revision
+		 */
+		$revision = isset( $_GET['revision'] ) ? absint( $_GET['revision'] ) : 0; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+		$revision_message = ! empty( $revision ) ?
+			sprintf(
+				/* translators: %s: Number of stories. */
+				__( 'Post restored to revision from %s.', 'web-stories' ),
```
Comment (swissspidy) [inline includes/Admin/Editor.php]: Nit:

```suggestion
				__( 'Story restored to revision from %s.', 'web-stories' ),
```

File: includes/Admin/Editor.php [hunk 3]
```diff
@@ -387,6 +387,20 @@ public function get_editor_settings(): array {
...
+		/**
+		 * Revision.
+		 *
+		 * @var int $revision
+		 */
+		$revision = isset( $_GET['revision'] ) ? absint( $_GET['revision'] ) : 0; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+		$revision_message = ! empty( $revision ) ?
+			sprintf(
+				/* translators: %s: Number of stories. */
```
Comment (swissspidy) [inline includes/Admin/Editor.php]:
```suggestion
				/* translators: %s: Date and time of the revision. */
```

**Review by swissspidy — APPROVED** [review]
Works like a charm!

**Pull request merged**